File: day-01/part-1.ts
import { loadInput } from "../utils";


const sum = (arr: number[]) => arr.reduce((acc, el) => acc + el, 0);

const input = loadInput("input");
const elfs = input.split("\n\n");

const cals = elfs.map(cals => cals.split("\n").map(cal => parseInt(cal, 10)));
const calsPerElf = cals.map(sum);

console.log(Math.max(...calsPerElf));

File: kotlin/src/main/kotlin/day01/part-1.kt
package day01.part1

import loadInput

fun main() {
    val input = loadInput("./src/main/kotlin/day01/input")

    val elfs = input.split("\n\n")
    val cals = elfs.map { it.split("\n").map { it.toInt() } }

    val total = cals.map { it -> it.sum() }
    println(total.max())
}
